src/Model_Hydro/CPU_Hydro/CPU_Shared_FluUtility.cpp

```diff
@@ -505,159 +505,228 @@ real Hydro_CheckMinEintInEngy( const real Dens, const real MomX, const real MomY
+         if ( Fields[0] <=   (real)0.0  ||  Fields[0] >= __FLT_MAX__  ||  Fields[0] !=  Fields[0] )
+            FailCell = true;
 
-   else if ( Cons == NULL && Prim != NULL && Input == NULL )
-   {
-//    check NaN
-      if (  Prim[0] != Prim[0]
-         || Prim[1] != Prim[1]
-         || Prim[2] != Prim[2]
-         || Prim[3] != Prim[3]
-         || Prim[4] != Prim[4]  )                                                         goto GOTCHA;
-
-//    check +inf and -inf for velocities and negative for mass density and pressure
-      if (  (real)  TINY_NUMBER >= Prim[0] || Prim[0]  >= (real)HUGE_NUMBER
-         || (real) -HUGE_NUMBER >= Prim[1] || Prim[1]  >= (real)HUGE_NUMBER
-         || (real) -HUGE_NUMBER >= Prim[2] || Prim[2]  >= (real)HUGE_NUMBER
-         || (real) -HUGE_NUMBER >= Prim[3] || Prim[3]  >= (real)HUGE_NUMBER
-         || (real)  TINY_NUMBER >= Prim[4] || Prim[4]  >= (real)HUGE_NUMBER )             goto GOTCHA;
-
-
-//    pass all checks
-      return false;
-   }
-   else
-   {
-      printf("ERROR: Two of Cons, Prim, and Input must be NULL !! at file <%s>, line <%d>, function <%s>\n",
-              File, Line, Function );
-      return true;
-    }
+         if ( FailCell && Verbose )
+            ( SingleFieldName )? printf( "ERROR: invalid %s (%14.7e) at file <%s>, line <%d>, function <%s>\n",
+                                          SingleFieldName, Fields[0], File, Line, Function ):
+                                 printf( "ERROR: invalid (%14.7e) at file <%s>, line <%d>, function <%s>\n",
+                                          Fields[0], File, Line, Function );
+         return FailCell;
 
-//  print all variables if goto GOTCHA
-    GOTCHA:
-    {
-      if ( Show )
-       {
+      break;
 
-         if ( Cons == NULL && Prim == NULL && Input != NULL && Info != NULL )
+
+      // === check conserved variables, including passive scalars ===
+      case UNPHY_MODE_CONS:
+
+
+         for (int v=0; v<NCOMP_TOTAL; v++)
          {
-            printf( "ERROR: invalid %s (%14.7e) at file <%s>, line <%d>, function <%s>\n",
-                     Info, *Input, File, Line, Function );
-#           if ( NCOMP_PASSIVE > 0 )
-            if ( Passive != NULL )
+            // check NaN
+            if ( Fields[v] != Fields[v] )
+                  FailCell = true;
+
+            // check momentum densities
+            if ( v == MOMX || v == MOMY || v == MOMZ )
             {
-              printf( "        Passive scalars:" );
-              for (int v=0; v<NCOMP_PASSIVE; v++)    printf( " %d=%13.7e", v, Passive[v] ); 
-              printf( "\n" );
+               if ( (real) Fields[v] <= -HUGE_NUMBER || Fields[v]  >= (real)HUGE_NUMBER )
+                  FailCell = true;
+            }
+            // otherwise, including passive scalars
+            else
+            {
+               if ( (real) Fields[v] <=  TINY_NUMBER || Fields[v]  >= (real)HUGE_NUMBER )
+                  FailCell = true;
             }
```

Comment:
Strictly speaking, passive scalars can be zero. So replace it with the following.
```
// check mass and energy densities
else if ( v < NCOMP_TOTAL ) {
   if ( Fields[v] <= TINY_NUMBER  ||  Fields[v] >= HUGE_NUMBER )  FailCell = true;
}  
            
// check passive scalars
else {
   if ( Fields[v] < (real)0.0  ||  Fields[v] >= HUGE_NUMBER )     FailCell = true;
}
```